docs/topics/control-flow.md

```diff
@@ -1,82 +1,96 @@
+to check within parentheses `()` and the action to take if the result is true within curly braces `{}`. You can use `else`
+and `else if` for additional branches and checks. 
+
+You can also write `if` as an expression, which lets you assign its returned value directly to a variable.
+If you use `if` as an expression, you must have an `else` branch.
+
+For example:
 
 ```kotlin
 fun main() {
-    val a = 2
-    val b = 3
+    val heightAlice = 160
+    val heightBob = 175
 
     //sampleStart
-    var max = a
-    if (a < b) max = b
+    var taller = heightAlice
+    if (heightAlice < heightBob) taller = heightBob
 
     // With else
-    if (a > b) {
-      max = a
+    if (heightAlice > heightBob) {
+        taller = heightAlice
     } else {
-      max = b
+        taller = heightBob
     }
 
     // As expression
-    max = if (a > b) a else b
+    taller = if (heightAlice > heightBob) heightAlice else heightBob
 
-    // You can also use `else if` in expressions:
-    val maxLimit = 1
-    val maxOrLimit = if (maxLimit > a) maxLimit else if (a > b) a else b
-  
-    println("max is $max")
-    // max is 3
-    println("maxOrLimit is $maxOrLimit")
-    // maxOrLimit is 3
+    // With `else if` as an expression:
+    val heightLimit = 150
+    val heightOrLimit = if (heightLimit > heightAlice) heightLimit else if (heightAlice > heightBob) heightAlice else heightBob
+
+    println("Taller height is $taller")
+    // Taller height is 175
+    println("Height or limit is $heightOrLimit")
+    // Height or limit is 175
     //sampleEnd
 }
 ```
 {kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="if-else-if-kotlin"}
 
-Branches of an `if` expression can be blocks. In this case, the last expression is the value of a block:
+Each branch in an `if` expression can be a block, where the value of the last expression becomes the result:
 
 ```kotlin
-val max = if (a > b) {
-    print("Choose a")
-    a
-} else {
-    print("Choose b")
-    b
+fun main() {
+    //sampleStart
+    val heightAlice = 160
+    val heightBob = 175
+
+    val taller = if (heightAlice > heightBob) {
+        print("Choose Alice\n")
+        heightAlice
+    } else {
+        print("Choose Bob\n")
+        heightBob
+    }
+
+    println("Taller height is $taller")
+    //sampleEnd
 }
 ```
-
-If you're using `if` as an expression, for example, for returning its value or
-assigning it to a variable, the `else` branch is mandatory.
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="if-else-blocks-kotlin"}
 
 ## When expressions and statements
 
-`when` is a conditional expression that runs code based on multiple possible values or conditions. It is
-similar to the `switch` statement in Java, C, and similar languages. For example:
+`when` is a conditional expression that runs code based on multiple possible values or conditions. It's
+similar to the `switch` statement in Java, C, and similar languages. `when` evaluates its argument against each branch 
```

Comment:
The word "evaluates" here is a bit dangerous, since this phrase could be read as if the argument was run multiple times, each per branch. Maybe "`when` evaluates its argument and compares the result against each branch ..."?